### PR TITLE
perf(database): Cache and reuse IDatabaseConnection across tests

### PR DESCRIPTION
**Pull request by jeffjensen**
## Summary by CodeRabbit

* **New Features**
  * Added opt-in, liveness-validated `CachingConnectionProvider` for connection reuse across testers.
  * Added shared-testcase connection support via `closeConnectionAfterTest` (default: close).
* **Bug Fixes**
  * Improved failure handling by preserving the primary failure and attaching cleanup/suppressed failures.
  * Hardened locale handling (Turkish-safe) and pinned encoding/charset behavior to UTF-8; improved handling of supplementary/control characters in XML exports.
  * Fixed several dataset/export correctness edge cases (CSV/XML/YAML/DTD/Oracle XML) plus transaction rollback, refresh cleanup, and safer parsing/stream/file handling.
* **Documentation**
  * Updated FAQ with safe connection-reuse guidance.
* **Tests**
  * Added/expanded coverage for reuse, lifecycle, locale, and encoding/export regressions.
* **Chores**
  * Bumped version to **3.4.0-SNAPSHOT**.

### CODE REVIEW

File: src/changes/changes.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
...
+    <release version="3.3.1-SNAPSHOT" date="TBD" description="Un-skip and strengthen remaining disabled and no-op tests: implement DefaultPrepAndExpectedTestCaseTest stubs, convert ForwardOnlyDataSetTest no-ops to assertThrows, implement ExcludeTableFilterTest reverse-order tests, relocate ToStringViewTest's package-blocked cases, restore commented-out unit tests in CsvParserTest/FlatXmlTableTest/BiDirectionalEdgesDepthFirstSearchTest, implement ReplacementTableTest.testGetMissingValue, add assertions to applyColumnFilters tests, fix DatabaseTestCase tear-down exceptions obscuring test failures, rewrite obsolete DbUnitTaskIT queryset test, delete two dead commented-out test fragments, stop swallowing failures in PostgreSQL xtest* methods, add CachingConnectionProvider for opt-in IDatabaseConnection reuse across test methods, reduce DefaultPrepAndExpectedTestCase's per-test connection churn">
```
[the remark this answers is not in the document]
Comment (jeffjensen) [reply src/changes/changes.xml]: Leaving this as-is: the long-changelog style for the release `description` attribute is this project's established practice, not an oversight here. The previous release entry (3.3.0) uses the identical pattern — a comma-separated summary of every change in the release, e.g. `description="Performance improvements and fixes: speed up column index lookups, suppress RowOutOfBoundsException stack traces, buffer XML, CSV, DTD, and YAML output, ..."` — and is, if anything, longer than this one. The per-`<action>` entries already carry the detailed prose for each individual change; the release-level `description` serves as this project's site-generated "at a glance" summary of the whole release rather than a one-line blurb, so keeping it comprehensive here is intentional and consistent with prior releases.

File: src/test/java/org/dbunit/DataSourceDatabaseTesterTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit tests for {@link DataSourceDatabaseTester}, run against a real H2 in-memory
+ * {@link DataSource}, focused on its opt-in {@link CachingConnectionProvider} integration.
+ *
+ * @since 3.3.1
+ */
+class DataSourceDatabaseTesterTest
```
[the remark this answers is not in the document]
Comment (jeffjensen) [reply src/test/java/org/dbunit/DataSourceDatabaseTesterTest.java]: done

**Comment by jeffjensen** [comment]
@coderabbitai resume

File: src/main/java/org/dbunit/DefaultPrepAndExpectedTestCase.java [hunk 3]
```diff
@@ -163,23 +228,103 @@ public void configureTest(
...
         } finally
         {
-            if (connection != null)
+            if (acquiredConnectionHere)
             {
-                connection.close();
+                closeReusableConnection();
             }
         }
+    }
```
[the remark this answers is not in the document]
Comment (jeffjensen) [reply src/main/java/org/dbunit/DefaultPrepAndExpectedTestCase.java:250]: Fixed: `lookupFeatureValue()` and `cleanupData()` now close the reusable connection on the success path and, on a primary failure, attach any close failure to it via a new `closeReusableConnectionSuppressing()` helper (using `Throwable#addSuppressed`) instead of letting the close exception replace and hide the primary - matching `runTest()` and `DatabaseTestCase.tearDown(Throwable)`. This is in a `fixup!` commit for the #800 perf commit, to be squashed in later.

_Addressed via Claude Code._

**Comment by jeffjensen** [comment]
Addressed the `.as()` nitpick from the review summary: `CachingConnectionProviderIT`'s connection-identity assertion in `testClose_thenGetConnection_reconnectsSuccessfully()` now reads:

```java
assertThat(second)
        .as("Closing the provider must create a new connection.")
        .isNotSameAs(first);
```

This is in a `fixup!` commit for the #799 feat commit, to be squashed in later.

_Addressed via Claude Code._

File: src/main/java/org/dbunit/DefaultPrepAndExpectedTestCase.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffjensen) [reply src/main/java/org/dbunit/DefaultPrepAndExpectedTestCase.java:250]: done

**Comment by jeffjensen** [comment]
Addressed the 5 Turkish-locale `Locale.setDefault()` duplication nitpicks from the latest review (`OracleConnectionTest`, `OracleSdoGeometryDataTypeTest`, `TruncateTableOperationTest`, `BytesDataTypeTest`, plus the broader "centralize across six tests" note).

Introduced a `@TurkishDefaultLocale` JUnit 5 extension (`TurkishLocaleExtension`) in `src/test/java/org/dbunit/` and applied it to all 13 test classes using the manual save/mutate/restore pattern - the 9 named across the two reviews plus 4 more with the identical pattern (`PatternMatcherTest`, `DbUnitAssertBaseTest`, `TablesDependencyHelperTest`, `AbstractTableMetaDataTest`, `LowerCaseDataSetTest`, `CaseInsensitiveDataSetTest`, `DefaultPrepAndExpectedTestCaseTest`, `FlatXmlProducerTest`, `DatabaseTableMetaDataIT`) - for full consistency rather than a partial fix.

Considered `junit-pioneer`'s `@DefaultLocale` first, but its latest release (2.3.0, Oct 2024) targets JUnit 5.11.2, a major version behind this project's Jupiter 6.1.1 with no newer release since, so went with a small custom extension instead to avoid that compatibility risk.

Tracked as dbunit/dbunit-extension#833. Commit 29e3e3e1.

_Addressed via Claude Code._

**Comment by jeffjensen** [comment]
@coderabbitai resume

File: src/main/java/org/dbunit/dataset/datatype/AbstractDataType.java [hunk 4]
```diff
@@ -200,6 +200,7 @@ public Object getSqlValue(final int column, final ResultSet resultSet)
     {
         logger.debug("getSqlValue(column={}, resultSet={}) - start", column,
                 resultSet);
+
```
[the remark this answers is not in the document]
Comment (jeffjensen) [reply src/main/java/org/dbunit/dataset/datatype/AbstractDataType.java:203]: its how we want it
